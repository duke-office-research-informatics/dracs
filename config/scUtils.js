import React from "react";
import { ThemeProvider } from "styled-components";
import { mount, shallow } from "enzyme";
import { theme as defaultTheme } from "../lib/dracs.es.js";

export const shallowWithTheme = (children, customTheme) => {
  const theme = customTheme || defaultTheme;
  return shallow(children, { context: theme });
};

/* eslint-disable */
export function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
    ),
  });
}

//FOR ACCESSING PORTALS VIA ENZYME -- import portal from dracs
// https://github.com/airbnb/enzyme/issues/1202#issuecomment-351639783
// function mountPortal(el){
//   const wrapper =  mountWithTheme(el);
//   wrapper.setState({ visible: true});
//   const portalInstance = wrapper.find(Portal).instance();
//   return mountWithTheme(portalInstance.props.children);
// }

//FOR ACCESSING EVENT LISTENERS ATTACHED TO DOCUMENT OR DOCUMENT.BODY
//https://github.com/airbnb/enzyme/issues/426#issuecomment-296304015
// const map = {};
// document.body.addEventListener = jest.fn((event, cb) => {
//   map[event] = cb;
// });
