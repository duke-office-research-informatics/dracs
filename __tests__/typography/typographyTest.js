import React from "react";
import { mountWithTheme } from "../../config/scUtils.js";
import { axe } from "../../config/axe-helper.js";
import { H1, H2, H3, H4, H5, P, theme } from "../../lib/dracs.es.js";

describe("H1", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mountWithTheme(<H1>An H1 is you</H1>);
  });

  it("matches dom snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("returns no accessiblity violations", async () => {
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  it("font-family: Roboto, Open-Sans, Sans-Serif", () => {
    expect(wrapper).toHaveStyleRule(
      "font-family",
      '"Roboto","Open-Sans",Sans-Serif'
    );
  });

  it("font-size: 2.625em", () => {
    expect(wrapper).toHaveStyleRule("font-size", "2.625em");
  });

  it("line-height: 1.16em", () => {
    expect(wrapper).toHaveStyleRule("line-height", "1.16em");
  });

  it("padding: 0", () => {
    expect(wrapper).toHaveStyleRule("padding", "0");
  });

  it("displays correct color when `color` prop is declared", () => {
    const el = mountWithTheme(
      <H1 color={theme.colors.action}>An H1 is you</H1>
    );
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("color", theme.colors.action);
  });

  it("sets the top/bottom margin when `margin` prop is declared", () => {
    const el = mountWithTheme(<H1 margin={8}>An H1 is you</H1>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("margin", "8px 0");
  });

  it("bolds when `bold` prop is set to true", () => {
    const el = mountWithTheme(<H1 bold>An H1 is you</H1>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("font-weight", "700");
  });

  it("italicizes when `italic` prop is set to true", () => {
    const el = mountWithTheme(<H1 italic>An H1 is you</H1>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("font-style", "italic");
  });
});

describe("H2", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mountWithTheme(<H2>An H2 is you</H2>);
  });

  it("matches dom snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("returns no accessiblity violations", async () => {
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  it("font-family: Roboto, Open-Sans, Sans-Serif", () => {
    expect(wrapper).toHaveStyleRule(
      "font-family",
      '"Roboto","Open-Sans",Sans-Serif'
    );
  });

  it("font-size: 2.125em", () => {
    expect(wrapper).toHaveStyleRule("font-size", "2.125em");
  });

  it("line-height: 1.18em", () => {
    expect(wrapper).toHaveStyleRule("line-height", "1.18em");
  });

  it("padding: 0", () => {
    expect(wrapper).toHaveStyleRule("padding", "0");
  });

  it("displays correct color when `color` prop is declared", () => {
    const el = mountWithTheme(
      <H2 color={theme.colors.action}>An H2 is you</H2>
    );
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("color", theme.colors.action);
  });

  it("sets the top/bottom margin when `margin` prop is declared", () => {
    const el = mountWithTheme(<H2 margin={8}>An H2 is you</H2>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("margin", "8px 0");
  });

  it("bolds when `bold` prop is set to true", () => {
    const el = mountWithTheme(<H2 bold>An H2 is you</H2>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("font-weight", "700");
  });

  it("italicizes when `italic` prop is set to true", () => {
    const el = mountWithTheme(<H2 italic>An H2 is you</H2>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("font-style", "italic");
  });
});

describe("H3", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mountWithTheme(<H3>An H3 is you</H3>);
  });

  it("matches dom snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("returns no accessiblity violations", async () => {
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  it("font-family: Roboto, Open-Sans, Sans-Serif", () => {
    expect(wrapper).toHaveStyleRule(
      "font-family",
      '"Roboto","Open-Sans",Sans-Serif'
    );
  });

  it("font-size: 1.5em", () => {
    expect(wrapper).toHaveStyleRule("font-size", "1.5em");
  });

  it("line-height: 1.35em", () => {
    expect(wrapper).toHaveStyleRule("line-height", "1.35em");
  });

  it("padding: 0", () => {
    expect(wrapper).toHaveStyleRule("padding", "0");
  });

  it("displays correct color when `color` prop is declared", () => {
    const el = mountWithTheme(
      <H3 color={theme.colors.action}>An H3 is you</H3>
    );
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("color", theme.colors.action);
  });

  it("sets the top/bottom margin when `margin` prop is declared", () => {
    const el = mountWithTheme(<H3 margin={8}>An H3 is you</H3>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("margin", "8px 0");
  });

  it("bolds when `bold` prop is set to true", () => {
    const el = mountWithTheme(<H3 bold>An H3 is you</H3>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("font-weight", "700");
  });

  it("italicizes when `italic` prop is set to true", () => {
    const el = mountWithTheme(<H3 italic>An H3 is you</H3>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("font-style", "italic");
  });
});

describe("H4", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mountWithTheme(<H4>An H4 is you</H4>);
  });

  it("matches dom snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("returns no accessiblity violations", async () => {
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  it("font-family: Roboto, Open-Sans, Sans-Serif", () => {
    expect(wrapper).toHaveStyleRule(
      "font-family",
      '"Roboto","Open-Sans",Sans-Serif'
    );
  });

  it("font-size: 1em", () => {
    expect(wrapper).toHaveStyleRule("font-size", "1em");
  });

  it("line-height: 1.5em", () => {
    expect(wrapper).toHaveStyleRule("line-height", "1.5em");
  });

  it("padding: 0", () => {
    expect(wrapper).toHaveStyleRule("padding", "0");
  });

  it("displays correct color when `color` prop is declared", () => {
    const el = mountWithTheme(
      <H4 color={theme.colors.action}>An H4 is you</H4>
    );
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("color", theme.colors.action);
  });

  it("sets the top/bottom margin when `margin` prop is declared", () => {
    const el = mountWithTheme(<H4 margin={8}>An H4 is you</H4>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("margin", "8px 0");
  });

  it("bolds when `bold` prop is set to true", () => {
    const el = mountWithTheme(<H4 bold>An H4 is you</H4>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("font-weight", "700");
  });

  it("italicizes when `italic` prop is set to true", () => {
    const el = mountWithTheme(<H4 italic>An H4 is you</H4>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("font-style", "italic");
  });
});

describe("H5", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mountWithTheme(<H5>An H5 is you</H5>);
  });

  it("matches dom snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("returns no accessiblity violations", async () => {
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  it("font-family: Roboto, Open-Sans, Sans-Serif", () => {
    expect(wrapper).toHaveStyleRule(
      "font-family",
      '"Roboto","Open-Sans",Sans-Serif'
    );
  });

  it("font-size: 0.875em", () => {
    expect(wrapper).toHaveStyleRule("font-size", "0.875em");
  });

  it("line-height: 1.15em", () => {
    expect(wrapper).toHaveStyleRule("line-height", "1.15em");
  });

  it("padding: 0", () => {
    expect(wrapper).toHaveStyleRule("padding", "0");
  });

  it("line-height: 1.43em when `expanded` is set to true", () => {
    const el = mountWithTheme(<H5 expanded>An H5 is you</H5>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("line-height", "1.43em");
  });

  it("displays correct color when `color` prop is declared", () => {
    const el = mountWithTheme(
      <H5 color={theme.colors.action}>An H5 is you</H5>
    );
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("color", theme.colors.action);
  });

  it("sets the top/bottom margin when `margin` prop is declared", () => {
    const el = mountWithTheme(<H5 margin={8}>An H5 is you</H5>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("margin", "8px 0");
  });

  it("bolds when `bold` prop is set to true", () => {
    const el = mountWithTheme(<H5 bold>An H5 is you</H5>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("font-weight", "700");
  });

  it("italicizes when `italic` prop is set to true", () => {
    const el = mountWithTheme(<H5 italic>An H5 is you</H5>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("font-style", "italic");
  });
});

describe("P", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mountWithTheme(<P>An P is you</P>);
  });

  it("matches dom snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("returns no accessiblity violations", async () => {
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  it("font-family: Roboto, Open-Sans, Sans-Serif", () => {
    expect(wrapper).toHaveStyleRule(
      "font-family",
      '"Roboto","Open-Sans",Sans-Serif'
    );
  });

  it("font-size: 0.875em", () => {
    expect(wrapper).toHaveStyleRule("font-size", "0.875em");
  });

  it("line-height: 1.15em", () => {
    expect(wrapper).toHaveStyleRule("line-height", "1.15em");
  });

  it("padding: 0", () => {
    expect(wrapper).toHaveStyleRule("padding", "0");
  });

  it("line-height: 1.43em when `expanded` is set to true", () => {
    const el = mountWithTheme(<P expanded>An P is you</P>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("line-height", "1.43em");
  });

  it("displays correct color when `color` prop is declared", () => {
    const el = mountWithTheme(<P color={theme.colors.action}>An P is you</P>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("color", theme.colors.action);
  });

  it("sets the top/bottom margin when `margin` prop is declared", () => {
    const el = mountWithTheme(<P margin={8}>An P is you</P>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("margin", "8px 0");
  });

  it("bolds when `bold` prop is set to true", () => {
    const el = mountWithTheme(<P bold>An P is you</P>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("font-weight", "700");
  });

  it("italicizes when `italic` prop is set to true", () => {
    const el = mountWithTheme(<P italic>An P is you</P>);
    expect(el).toMatchSnapshot();
    expect(el).toHaveStyleRule("font-style", "italic");
  });
});
