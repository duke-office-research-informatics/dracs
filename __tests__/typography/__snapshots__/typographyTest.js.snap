// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 bolds when \`bold\` prop is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 2.625em;
  line-height: 1.16em;
  color: #131516;
  font-weight: 700;
  margin: 4px 0;
  padding: 0;
}

<typography__H1
  bold={true}
>
  <h1
    className="c0"
  >
    An H1 is you
  </h1>
</typography__H1>
`;

exports[`H1 displays correct color when \`color\` prop is declared 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 2.625em;
  line-height: 1.16em;
  color: #0680cd;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__H1
  color="#0680cd"
>
  <h1
    className="c0"
    color="#0680cd"
  >
    An H1 is you
  </h1>
</typography__H1>
`;

exports[`H1 italicizes when \`italic\` prop is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 2.625em;
  line-height: 1.16em;
  color: #131516;
  font-weight: 400;
  font-style: italic;
  margin: 4px 0;
  padding: 0;
}

<typography__H1
  italic={true}
>
  <h1
    className="c0"
  >
    An H1 is you
  </h1>
</typography__H1>
`;

exports[`H1 matches dom snapshot 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 2.625em;
  line-height: 1.16em;
  color: #131516;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__H1>
  <h1
    className="c0"
  >
    An H1 is you
  </h1>
</typography__H1>
`;

exports[`H1 sets the top/bottom margin when \`margin\` prop is declared 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 2.625em;
  line-height: 1.16em;
  color: #131516;
  font-weight: 400;
  margin: 8px 0;
  padding: 0;
}

<typography__H1
  margin={8}
>
  <h1
    className="c0"
  >
    An H1 is you
  </h1>
</typography__H1>
`;

exports[`H2 bolds when \`bold\` prop is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 2.125em;
  line-height: 1.18em;
  color: #131516;
  font-weight: 700;
  margin: 4px 0;
  padding: 0;
}

<typography__H2
  bold={true}
>
  <h2
    className="c0"
  >
    An H2 is you
  </h2>
</typography__H2>
`;

exports[`H2 displays correct color when \`color\` prop is declared 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 2.125em;
  line-height: 1.18em;
  color: #0680cd;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__H2
  color="#0680cd"
>
  <h2
    className="c0"
    color="#0680cd"
  >
    An H2 is you
  </h2>
</typography__H2>
`;

exports[`H2 italicizes when \`italic\` prop is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 2.125em;
  line-height: 1.18em;
  color: #131516;
  font-weight: 400;
  font-style: italic;
  margin: 4px 0;
  padding: 0;
}

<typography__H2
  italic={true}
>
  <h2
    className="c0"
  >
    An H2 is you
  </h2>
</typography__H2>
`;

exports[`H2 matches dom snapshot 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 2.125em;
  line-height: 1.18em;
  color: #131516;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__H2>
  <h2
    className="c0"
  >
    An H2 is you
  </h2>
</typography__H2>
`;

exports[`H2 sets the top/bottom margin when \`margin\` prop is declared 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 2.125em;
  line-height: 1.18em;
  color: #131516;
  font-weight: 400;
  margin: 8px 0;
  padding: 0;
}

<typography__H2
  margin={8}
>
  <h2
    className="c0"
  >
    An H2 is you
  </h2>
</typography__H2>
`;

exports[`H3 bolds when \`bold\` prop is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1.5em;
  line-height: 1.35em;
  color: #131516;
  font-weight: 700;
  margin: 4px 0;
  padding: 0;
}

<typography__H3
  bold={true}
>
  <h3
    className="c0"
  >
    An H3 is you
  </h3>
</typography__H3>
`;

exports[`H3 displays correct color when \`color\` prop is declared 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1.5em;
  line-height: 1.35em;
  color: #0680cd;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__H3
  color="#0680cd"
>
  <h3
    className="c0"
    color="#0680cd"
  >
    An H3 is you
  </h3>
</typography__H3>
`;

exports[`H3 italicizes when \`italic\` prop is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1.5em;
  line-height: 1.35em;
  color: #131516;
  font-weight: 400;
  font-style: italic;
  margin: 4px 0;
  padding: 0;
}

<typography__H3
  italic={true}
>
  <h3
    className="c0"
  >
    An H3 is you
  </h3>
</typography__H3>
`;

exports[`H3 matches dom snapshot 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1.5em;
  line-height: 1.35em;
  color: #131516;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__H3>
  <h3
    className="c0"
  >
    An H3 is you
  </h3>
</typography__H3>
`;

exports[`H3 sets the top/bottom margin when \`margin\` prop is declared 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1.5em;
  line-height: 1.35em;
  color: #131516;
  font-weight: 400;
  margin: 8px 0;
  padding: 0;
}

<typography__H3
  margin={8}
>
  <h3
    className="c0"
  >
    An H3 is you
  </h3>
</typography__H3>
`;

exports[`H4 bolds when \`bold\` prop is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #545454;
  font-weight: 700;
  margin: 4px 0;
  padding: 0;
}

<typography__H4
  bold={true}
>
  <h4
    className="c0"
  >
    An H4 is you
  </h4>
</typography__H4>
`;

exports[`H4 displays correct color when \`color\` prop is declared 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #0680cd;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__H4
  color="#0680cd"
>
  <h4
    className="c0"
    color="#0680cd"
  >
    An H4 is you
  </h4>
</typography__H4>
`;

exports[`H4 italicizes when \`italic\` prop is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #545454;
  font-weight: 400;
  font-style: italic;
  margin: 4px 0;
  padding: 0;
}

<typography__H4
  italic={true}
>
  <h4
    className="c0"
  >
    An H4 is you
  </h4>
</typography__H4>
`;

exports[`H4 matches dom snapshot 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #545454;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__H4>
  <h4
    className="c0"
  >
    An H4 is you
  </h4>
</typography__H4>
`;

exports[`H4 sets the top/bottom margin when \`margin\` prop is declared 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #545454;
  font-weight: 400;
  margin: 8px 0;
  padding: 0;
}

<typography__H4
  margin={8}
>
  <h4
    className="c0"
  >
    An H4 is you
  </h4>
</typography__H4>
`;

exports[`H5 bolds when \`bold\` prop is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 0.875em;
  line-height: 1.15em;
  color: #696969;
  font-weight: 700;
  margin: 4px 0;
  padding: 0;
}

<typography__H5
  bold={true}
>
  <h5
    className="c0"
  >
    An H5 is you
  </h5>
</typography__H5>
`;

exports[`H5 displays correct color when \`color\` prop is declared 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 0.875em;
  line-height: 1.15em;
  color: #0680cd;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__H5
  color="#0680cd"
>
  <h5
    className="c0"
    color="#0680cd"
  >
    An H5 is you
  </h5>
</typography__H5>
`;

exports[`H5 italicizes when \`italic\` prop is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 0.875em;
  line-height: 1.15em;
  color: #696969;
  font-weight: 400;
  font-style: italic;
  margin: 4px 0;
  padding: 0;
}

<typography__H5
  italic={true}
>
  <h5
    className="c0"
  >
    An H5 is you
  </h5>
</typography__H5>
`;

exports[`H5 line-height: 1.43em when \`expanded\` is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 0.875em;
  line-height: 1.43em;
  color: #696969;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__H5
  expanded={true}
>
  <h5
    className="c0"
  >
    An H5 is you
  </h5>
</typography__H5>
`;

exports[`H5 matches dom snapshot 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 0.875em;
  line-height: 1.15em;
  color: #696969;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__H5>
  <h5
    className="c0"
  >
    An H5 is you
  </h5>
</typography__H5>
`;

exports[`H5 sets the top/bottom margin when \`margin\` prop is declared 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 0.875em;
  line-height: 1.15em;
  color: #696969;
  font-weight: 400;
  margin: 8px 0;
  padding: 0;
}

<typography__H5
  margin={8}
>
  <h5
    className="c0"
  >
    An H5 is you
  </h5>
</typography__H5>
`;

exports[`P bolds when \`bold\` prop is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 0.875em;
  line-height: 1.15em;
  color: #131516;
  font-weight: 700;
  margin: 4px 0;
  padding: 0;
}

<typography__P
  bold={true}
>
  <p
    className="c0"
  >
    An P is you
  </p>
</typography__P>
`;

exports[`P displays correct color when \`color\` prop is declared 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 0.875em;
  line-height: 1.15em;
  color: #0680cd;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__P
  color="#0680cd"
>
  <p
    className="c0"
    color="#0680cd"
  >
    An P is you
  </p>
</typography__P>
`;

exports[`P italicizes when \`italic\` prop is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 0.875em;
  line-height: 1.15em;
  color: #131516;
  font-weight: 400;
  font-style: italic;
  margin: 4px 0;
  padding: 0;
}

<typography__P
  italic={true}
>
  <p
    className="c0"
  >
    An P is you
  </p>
</typography__P>
`;

exports[`P line-height: 1.43em when \`expanded\` is set to true 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 0.875em;
  line-height: 1.43em;
  color: #131516;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__P
  expanded={true}
>
  <p
    className="c0"
  >
    An P is you
  </p>
</typography__P>
`;

exports[`P matches dom snapshot 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 0.875em;
  line-height: 1.15em;
  color: #131516;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
}

<typography__P>
  <p
    className="c0"
  >
    An P is you
  </p>
</typography__P>
`;

exports[`P sets the top/bottom margin when \`margin\` prop is declared 1`] = `
.c0 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 0.875em;
  line-height: 1.15em;
  color: #131516;
  font-weight: 400;
  margin: 8px 0;
  padding: 0;
}

<typography__P
  margin={8}
>
  <p
    className="c0"
  >
    An P is you
  </p>
</typography__P>
`;
